symengine/solve.cpp

```diff
@@ -635,13 +632,12 @@ linear_eqns_to_matrix(const vec_basic &equations, const vec_sym &syms)
             if (not non_zero) {
                 rem = res;
             } else {
-                coeffs[index_of_sym[cursim]] = res;
```

Comment:
@certik, this vector `coeffs` was only set if the coefficients were non-zero. `x+ 2*y` was fine, `x` was not.
A solution was to start with a zero vector. In line 600, this is what is done.